pkg/providers/tinkerbell/pbnj/client.go

```diff
@@ -0,0 +1,60 @@
+package pbnj
+
+import (
+	"context"
+	"os"
+
+	pbnjClient "github.com/tinkerbell/cluster-api-provider-tinkerbell/pbnj/client"
+	pbnjv1 "github.com/tinkerbell/pbnj/api/v1"
+
+	"github.com/aws/eks-anywhere/pkg/hardware"
+)
+
+const (
+	PbnjGrpcAuth = "PBNJ_GRPC_AUTHORITY"
+)
+
+type Pbnj struct {
+	*pbnjClient.PbnjClient
+}
+
+func NewPBNJClient(pbnjGrpcAuth string) (*Pbnj, error) {
+	os.Setenv(PbnjGrpcAuth, pbnjGrpcAuth)
+
+	conn, _ := pbnjClient.SetupConnection()
+
+	mClient := pbnjv1.NewMachineClient(conn)
+	tClient := pbnjv1.NewTaskClient(conn)
+	pbnjObj := &Pbnj{pbnjClient.NewPbnjClient(mClient, tClient)}
+
+	os.Unsetenv(PbnjGrpcAuth)
```

Comment:
why is this needed?

Reply:
Actually that's needed because SetupConnection looks for `PBNJ_GRPC_AUTHORITY` env var for getting the IP and port to connect to.